### PR TITLE
Adds Food Allergies as a counterpart to Medicine Allergies

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

Adds food allergies as a counterpart to medicinal allergies. 

This is about as you would expect. Eating food you are allergic to causes Anaphylaxis, which can be cured with epinephrine. If you are not cured rapidly, you will shortly fall into a coma and die.

There is a preference for choosing your character's allergy. 
Not all food types are viable allergen types: Only `"Alcohol", "Bugs", "Dairy", "Fruit", "Grain", "Meat", "Nuts", "Seafood", "Sugar", "Vegetables"`.

You also get a dogtag like med allergies. 

Minor implementation detail: I added a "dangerous" level of food quality / scale, so I gave it to the Death Sandwich. 

## Why It's Good For The Game

This was a request, and I saw it fairly harmless, so I said why not. 

Similar to Medicine Allergy, it adds a small mechanic to concern yourself with when you visit the kitchen, encouraging you to get a bit more familiar with food types before diving into food at the kitchen. 

As it is not as impactful as Medicine Allergy, it is only a -2 quirk. 

## Changelog

:cl: Melbert
add: Adds Food Allergies as a -2 quirk. You can select which food you're allergic to or rock a random option. 
/:cl:

### CODE REVIEW

**Comment by Wallemations** [comment]
Having bugs on there tells me we need more bug foods
Boutta make the giant ant butcherable for ant meat

**Comment by xXPawnStarrXx** [comment]
Yeah. I'm tempted to make chocolate a food type too, so you can kill Ian with it.

File: code/game/objects/items/food/sandwichtoast.dm [hunk 1]
```diff
@@ -284,7 +284,7 @@
 	// Closest thing to a mullet we have
 	if(consumer.hairstyle == "Gelled Back" && istype(consumer.get_item_by_slot(ITEM_SLOT_ICLOTHING), /obj/item/clothing/under/rank/civilian/cookjorts))
 		return FOOD_LIKED
-	return FOOD_DISLIKED
+	return FOOD_ALLERGIC
```
Comment (Ghommie) [inline code/game/objects/items/food/sandwichtoast.dm:287]: You already get a specific disease when you eat it without meeting the requirements, so why?

**Comment by YakumoChen** [comment]
Alcohol is not a food type widely used, its only in two items (filet minrawr and one other one)

Bar drinks don't have a food type because it's based on the container, not the contents.

**Comment by CRITAWAKETS** [comment]
If alcohol is given to the drink variant, then I feel powergamers will likely just take this quirk with Bugs or Seafood, since those tend to be not too common for food types, meaning free -2.

A _funnier_ way would be to make this work via chemicals instead so a crafty chemist could figure out their target's allergy and then give them their allergen.

**Comment by YakumoChen** [comment]
We'd have to recode a lot of the food system I think if food types contain corresponding chemicals, fairly out of scope

I'd remove bugs and alcohol since they're rare anyway

File: code/game/objects/items/food/sandwichtoast.dm (hunk 1 above)
Comment (MrMelbert) [reply code/game/objects/items/food/sandwichtoast.dm:287]: Quote from PR 

> Minor implementation detail: I added a "dangerous" level of food quality / scale, so I gave it to the Death Sandwich.

Food allergic is just synonymous with "this food will straight up kill you", it doesn't actually give you anaphylaxis unless you actually have an allergy. 

If you'd like me to rename it though

**Comment by D4C-420** [comment]
Touching something your allergic to should make your sprite swell and cause itching.

**Pull request merged**